src/app/client/src/app/modules/dashboard/services/report/report.service.ts

```diff
@@ -0,0 +1,123 @@
+import { Injectable } from '@angular/core';
+import { DomSanitizer } from '@angular/platform-browser';
+import { UsageService } from '../usage/usage.service';
+import { map } from 'rxjs/operators';
+import * as _ from 'lodash-es';
+import { Observable } from 'rxjs';
+
+@Injectable()
+export class ReportService {
+
+  constructor(private sanitizer: DomSanitizer, private usageService: UsageService, private userService: UserService,
+    private configService: ConfigService, private baseReportService: BaseReportService, private permissionService: PermissionService) { }
+
+  public fetchDataSource(filePath: string): Observable<any> {
+    return this.usageService.getData(filePath).pipe(
+      map(configData => {
+        return _.get(configData, 'result');
+      })
+    );
+  }
+
+  public fetchReportById(id): Observable<any> {
+    const req = {
+      url: `${this.configService.urlConFig.URLS.REPORT.READ}/${id}`
+    };
+    return this.baseReportService.get(req).pipe(
+      map(apiResponse => _.get(apiResponse, 'result'))
+    );
+  }
+
+  public listAllReports(filters: IListReportsFilter = {}) {
+    const request = {
+      url: this.configService.urlConFig.URLS.REPORT.LIST,
+      data: {
+        'request': {
+          'filters': filters
+        }
+      }
+    };
+    return this.baseReportService.post(request).pipe(
+      map(apiResponse => _.get(apiResponse, 'result'))
+    );
+  }
+
+  public prepareChartData(chartsArray: Array<any>, data: any, downloadUrl: string): Array<{}> {
+    return _.map(chartsArray, chart => {
+      const chartObj: any = {};
+      chartObj.chartConfig = chart;
+      chartObj.downloadUrl = downloadUrl;
+      chartObj.chartData = _.get(data, 'data');
+      chartObj.lastUpdatedOn = _.get(data, 'metadata.lastUpdatedOn');
+      return chartObj;
+    });
+  }
+
+  public prepareTableData(tablesArray: any, data: any, downloadUrl: string): Array<{}> {
+    tablesArray = _.isArray(tablesArray) ? tablesArray : [tablesArray];
+    return _.map(tablesArray, table => {
+      const tableData: any = {};
+      tableData.id = _.get(table, 'id') || `table-${_.random(1000)}`;
+      tableData.name = _.get(table, 'name') || 'Table';
+      tableData.header = _.get(table, 'columns') || _.get(data, _.get(table, 'columnsExpr'));
+      tableData.data = _.get(table, 'values') || _.get(data, _.get(table, 'valuesExpr'));
+      tableData.downloadUrl = _.get(table, 'downloadUrl') || downloadUrl;
+      return tableData;
+    });
+  }
+
+  public downloadReport(filePathSource) {
+    return this.fetchDataSource(filePathSource).pipe(
+      map(response => _.get(response, 'signedUrl'))
+    );
+  }
+
+  /**
+   * @description Report Viewer are only allowed to view these pages.
+   * @param {(string | undefined)} roles
+   * @returns {Observable<boolean>}
+   * @memberof ReportService
+   */
+  public isAuthenticated(roles: string | undefined): Observable<boolean> {
```

Comment:
can we throwError instead of returning true or false. This avoids throwing errorn in components and can be caught in catch block in components.

Reply:
That's not required as such. Bcz it's being used in two components reports.component and list-all-reports and I'm handling the error there.

Reply:
it won't go in the map block if the error occurs.  if an error occurs handling the error by returning false